deep_qa/data/instances/mc_tuples_with_background_instance.py

```diff
@@ -0,0 +1,358 @@
+            answer_tuples.append(curr_answer_tuples)
+        assert len(answer_tuples) == num_answers, \
+            "ERROR: {0} answers found, {1} specified.".format(len(answer_tuples), num_answers)
+        # Background tuples
+        background_tuple_strings = fields[3].split("$$$")
+        background_tuples = [TextTuple.from_string(background_string)
+                             for background_string in background_tuple_strings if len(background_string) > 0]
+        # Label
+        label = int(fields[4])
+        assert label < num_answers
+
+        return cls(answer_tuples, background_tuples, label=label, index=index)
+
+
+class IndexedMCTuplesWithBackgroundInstance(IndexedInstance):
+
+    def __init__(self,
+                 answers_indexed: List[List[List[List[int]]]],
+                 background_indexed: List[List[List[int]]],
+                 label,
+                 index: int=None):
+        super(IndexedMCTuplesWithBackgroundInstance, self).__init__(label, index)
+        self.answers_indexed = answers_indexed
+        self.background_indexed = background_indexed
+
+    @classmethod
+    @overrides
+    def empty_instance(cls):
+        return IndexedMCTuplesWithBackgroundInstance([], [], label=None, index=None)
+
+    @overrides
+    def get_lengths(self) -> Dict[str, int]:
+        # We care only about the longest slot here because all slots will be padded to the same length.
+        max_answer_length = 0
+        max_num_slots = 0
+        max_question_tuples = 0
+        for curr_answer in self.answers_indexed:
+            curr_max_num_slots, curr_max_answer_length = self.get_lengths_from_indexed_tuples(curr_answer)
+            max_num_slots = max(max_num_slots, curr_max_num_slots)
+            max_answer_length = max(max_answer_length, curr_max_answer_length)
+            max_question_tuples = max(max_question_tuples, len(curr_answer))
+        background_max_num_slots, background_max_slot_length = \
+            self.get_lengths_from_indexed_tuples(self.background_indexed)
+        max_num_slots = max(max_num_slots, background_max_num_slots)
+
+        return {'num_options': len(self.answers_indexed),
+                'num_question_tuples': max_question_tuples,
+                'num_background_tuples': len(self.background_indexed),
+                'question_slot_length': max_answer_length,
+                'background_slot_length': background_max_slot_length,
+                'num_slots': max_num_slots}
+
+    # Helper method for self.get_lengths() -- gets the max lengths for a list of indexed tuples
+    @staticmethod
+    def get_lengths_from_indexed_tuples(indexed_tuples: List[List[List[int]]]):
+        max_num_slots = 0
+        max_slot_length = 0
+        for indexed_tuple in indexed_tuples:
+            num_slots = len(indexed_tuple)
+            max_num_slots = max(max_num_slots, num_slots)
+            for slot in indexed_tuple:
+                max_slot_length = max(max_slot_length, len(slot))
```

Comment:
You need to call `self._get_word_sequence_lengths` here, to handle the case where the model using this `Instance` is using the `WordAndCharacter` tokenizer.  It'll make the return value and the logic in the method that calls this a little more complicated.